### ISSUE TITLE
[Settings import] Validate 'folderPushMode' value

### ISSUE THREAD

**Issue by cketti** [thunderbird/thunderbird-android#5363]
Make sure `folderPushMode` is set to `NONE` when importing settings of an account type that doesn't support Push.

**Comment by artbristol** [comment]
@cketti Just to clarify,  an account that doesn't support push = anything non-IMAP i.e. WebDav, JMAP (for now), POP3?

**Comment by cketti** [comment]
Yes. But I'd prefer if we don't duplicate the current logic. Right now there's `Backend.isPushCapable`. Maybe the mechanism needs to be changed so we can have `BackendManager.isPushCapable(serverType: String)`.

**Comment by artbristol** [comment]
I tried injecting the `BackendManager` into the `AccountPreferenceSerializer` but I ran into a stack overflow which I assume is because of a new circular dependency between app/k9mail and app/core. Any thoughts?

**Comment by cketti** [comment]
The validation logic lives inside the `Settings` class. You shouldn't have to touch `AccountPreferenceSerializer`.

**Comment by artbristol** [comment]
I will have the same problem if I try to inject `BackendManager` into `Settings` won't I? Since `Settings`, like `AccountPreferenceSerializer`, also lives in app/core.

Second question: all the validation in Settings seems to be single-field with no other context, i.e. if you have just the setting value and the type, you can tell whether the value is valid. But checking folderPushMode is a cross-field validation since you also need to know the setting for incoming server type. I'm happy to add a general cross-field mechanism, or just add a check for this field (I'd put it in `AccountSettingsDescriptions#validate`). What do you think?
